bignum.cc

```diff
@@ -177,6 +177,64 @@ int BN_jacobi_priv(const BIGNUM *A,const BIGNUM *N,int *jacobi,
   return returnvalue;
 }
 
+static const char ALPHANUMERIC[] = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+char *BN_bn2base36(const BIGNUM *a)
+{
+  int i = 0, num, ok = 0;
+  char *buf = NULL;
+  char *p;
+  BIGNUM *t = NULL;
+  BN_ULONG *bn_buff, *lp;
+
+  i = BN_num_bits(a);
+#define BN_BASE36_NUM 5 // log(36)
+
+  num = (i + BN_BASE36_NUM + 1) / BN_BASE36_NUM;
+  bn_buff = (BN_ULONG*) OPENSSL_malloc(num * sizeof(BN_ULONG));
+  buf = (char*) OPENSSL_malloc((num + 2) * sizeof(char));
```

Comment:
`sizeof(char)` is always 1 so multiplying by it is not that useful.

For the allocation on the line above, I'd use `sizeof(*bn_buff)`.  That way it keeps working when the type of `bn_buff` changes.